pkg/frontend/querymiddleware/stats.go

```diff
@@ -76,8 +76,10 @@ func (s queryStatsMiddleware) Do(ctx context.Context, req Request) (Response, er
 
 type QueryDetails struct {
 	*stats.Stats
-	Start, End time.Time
-	Step       time.Duration
+	Start, End           time.Time
+	Step                 time.Duration
+	UncachedResultsBytes int
+	CachedResultsBytes   int
```

Comment:
What about calling these `ResultCacheHitBytes` and `ResultCacheMissBytes` (and changing the names used on the log lines to match)? 

The current names are potentially ambiguous - it's not immediately clear if these are talking about reading from the cache or writing to the cache, or which cache this is referring to.